File: ETourBackend/src/main/java/com/example/demo/entity/Review.java
package com.tourism.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "review")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "review_id")
    private Integer reviewId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "booking_id")
    private Booking booking;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    private User customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tour_id")
    private Tour tour;

    @Column(name = "rating")
    private Integer rating;

    @Column(name = "review_title")
    private String reviewTitle;

    @Column(name = "review_description")
    private String reviewDescription;

    @Column(name = "review_date")
    private LocalDate reviewDate;

    @Column(name = "verification_status")
    private String verificationStatus;
}

File: ETour-DotNet/Models/Review.cs
namespace ETour.Api.Models;

public class Review
{
    public int ReviewId { get; set; }

    public int? BookingId { get; set; }
    public Booking Booking { get; set; }

    public int? CustomerId { get; set; }
    public User Customer { get; set; }

    public int? TourId { get; set; }
    public Tour Tour { get; set; }

    public int? Rating { get; set; }
    public string ReviewTitle { get; set; }
    public string ReviewDescription { get; set; }
    public DateOnly? ReviewDate { get; set; }
    public string VerificationStatus { get; set; }
}
